src/function/FunctionDraw.ts

```diff
@@ -1,98 +1,157 @@
-		}
-		elem.prop("attrs/labelAttrs/text", text.join("\n"));
-		const width = representation === Representation.COMPACT ?
-			Math.max((labels.reduce((a, b) => a.length > b.length ? a : b, "").length * 10) + 4,
-				text.length > 0 ? 8 * (text.reduce((a, b) => a.length > b.length ? a : b, "").length) : 0) :
-			(labels.reduce((a, b) => a.length > b.length ? a : b, "").length * 10) + 4;
-		elem.prop('attrs/text/x', width / 2);
-		const attrHeight = (24 + ((labels.length - 1) * 18));
-		const height = (text.length > 0 ? (4 + (text.length * 14)) : 0) +
-			attrHeight;
-		elem.prop('attrs/labelAttrs/y', attrHeight);
-		setElementShape(elem, width, height);
-		elem.resize(width, height);
-	}
+export function drawGraphElement(
+  elem: joint.dia.Element,
+  languageCode: string,
+  representation: number
+) {
+  if (typeof elem.id === "string") {
+    const types = WorkspaceTerms[WorkspaceElements[elem.id].iri].types;
+    setDisplayLabel(elem.id, languageCode);
+    const label = WorkspaceElements[elem.id].selectedLabel[languageCode];
+    const labels: string[] = [];
+    if (AppSettings.viewStereotypes)
+      getStereotypeList(types, languageCode).forEach((str) =>
+        labels.push("«" + str.toLowerCase() + "»")
+      );
+    labels.push(label === "" ? "<blank>" : label);
+    elem.prop("attrs/label/text", labels.join("\n"));
+    const text: string[] = [];
+    if (representation === Representation.COMPACT) {
+      Object.keys(WorkspaceLinks)
+        .filter(
+          (link) =>
+            WorkspaceLinks[link].active &&
+            WorkspaceLinks[link].source === elem.id &&
+            WorkspaceLinks[link].iri ===
+              parsePrefix("z-sgov-pojem", "má-vlastnost")
+        )
+        .forEach((link) =>
+          text.push(
+            WorkspaceTerms[WorkspaceElements[WorkspaceLinks[link].target].iri]
+              .labels[languageCode]
+          )
+        );
+      Object.keys(WorkspaceLinks)
+        .filter(
+          (link) =>
+            WorkspaceLinks[link].active &&
+            WorkspaceLinks[link].target === elem.id &&
+            WorkspaceLinks[link].iri ===
+              parsePrefix("z-sgov-pojem", "je-vlastností")
+        )
+        .forEach((link) =>
+          text.push(
+            WorkspaceTerms[WorkspaceElements[WorkspaceLinks[link].source].iri]
+              .labels[languageCode]
+          )
+        );
+    }
+    elem.prop("attrs/labelAttrs/text", text.join("\n"));
+    const width =
+      representation === Representation.COMPACT
+        ? Math.max(
+            labels.reduce((a, b) => (a.length > b.length ? a : b), "").length *
+              10 +
+              4,
+            text.length > 0
+              ? 8 *
+                  text.reduce((a, b) => (a.length > b.length ? a : b), "")
+                    .length
+              : 0
+          )
+        : labels.reduce((a, b) => (a.length > b.length ? a : b), "").length *
+            10 +
+          4;
```

Comment:
A ternary operator as a function argument, within a ternary operator... not readable at all. The reducer function is redundant.